dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/internal/ApplicationRegistryImpl.java

```diff
@@ -66,13 +66,22 @@ public class ApplicationRegistryImpl implements ApplicationRegistry {
     @Reference
     private MergeProcessor mergeProcessor;
 
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.STATIC, unbind = "unbindAppConfigListener")
+    public void bindAppConfigListener(OpenAPIAppConfigProvider openAPIAppConfigProvider) {
+        openAPIAppConfigProvider.registerAppConfigListener(this);
+    }
+
+    public void unbindAppConfigListener(OpenAPIAppConfigProvider openAPIAppConfigProvider) {
+        openAPIAppConfigProvider.unregisterAppConfigListener(this);
+    }
```

Comment:
I've noticed that this is a static reference.

We need to test what happens if the `<mpOpenAPI>` config element is removed altogether.

The component is set to optional configuration, but we should check whether that component ever gets deactivated since that would deactivate us and could cause lots of things to restart.

Reply:
I tested manually with osgiConsole and it looks like we still have OpenAPIAppConfigProvider. This is from after I removed the tag from a running server: 

```
io.openliberty.microprofile.openapi.internal.common.services.OpenAPIAppConfigProvider$OpenAPIAppConfigListener}={service.id=359, service.bundleid=130, service.scope=bundle, osgi.ds.satisfying.condition.target=(osgi.condition.id=true), component.name=io.openliberty.microprofile.openapi20.internal.ModuleSelectionConfigImpl, component.id=260}
{org.eclipse.microprofile.openapi.spi.OASFactoryResolver}={service.id=360, service.bundleid=130, service.scope=bundle, osgi.ds.satisfying.condition.target=(osgi.condition.id=true), service.vendor=IBM, component.name=io.openliberty.microprofile.openapi20.internal.OLOASFactoryResolverImpl, component.id=261}
```

Reply:
I'm not concerned about the `OpenAPIAppConfigProvider` being there, a lot of our test cases have no mpOpenAPI config, I'm concerned about whether it causes all the components to be restarted.

Reply:
We can note it down as something to check. Once we have a test which dynamically adds or removes the config, we can check the trace to see if the components are being deactivated and reactivated.